e2e/transform/transform-testrunner/test-runner.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {Config} from '@jest/types';
+import {JestEnvironment} from '@jest/environment';
+import Runtime from 'jest-runtime';
+import {TestResult} from '@jest/test-result';
+
+export default function testRunner(
+  globalConfig: Config.GlobalConfig,
+  config: Config.ProjectConfig,
+  environment: JestEnvironment,
+  runtime: Runtime,
+  testPath: string
+): Promise<TestResult> {
+  return Promise.resolve({
+    numFailingTests: 0,
+    numPassingTests: 1,
+    numPendingTests: 0,
+    numTodoTests: 0,
+    snapshot: {
+      added: 0,
+      fileDeleted: false,
+      matched: 0,
+      unchecked: 0,
+      unmatched: 0,
+      updated: 0,
+    },
+    testFilePath: testPath,
+    testResults: [
+      {
+        ancestorTitles: [],
+        duration: 2,
+        failureMessages: [],
+        fullName: 'sample test',
+        status: 'passed',
+        title: 'sample test',
+      },
+    ],
+  } as TestResult);
```

Comment:
Do you actually need this? TS should be able to infer the type based on the return.

Reply:
I think it might be needed.
To create that response I ran a test, printed the test result generated by `jest-jasmine2`.

It doesn't seem to have all the required fields in `TestResult`, the following are missing  `leaks, openHandles, skipped`

I can see what you mean,  might be some of those fields can actually be marked optional in the type?

Reply:
In `jest-jasmine2` some of the fields are missing

https://github.com/facebook/jest/blob/d610c9a00f5b9ca608ff9a7fb1e0a2b21f5f9027/packages/jest-jasmine2/src/reporter.ts#L80-L83

In `jest-circus` all the fields of `TestResult` are present

https://github.com/facebook/jest/blob/d610c9a00f5b9ca608ff9a7fb1e0a2b21f5f9027/packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapterInit.ts#L217-L220

I'll make the change according to whichever is correct

Reply:
@SimenB any suggestions? You mentioned to probably create a helper to "empty" test results. If so, currently the `TestResult` returned in jest-circus looks like the viable candidate to use as a template